registry/donor/models.py

```diff
@@ -84,6 +84,26 @@ class DonorsOverview(db.Model):
     def __repr__(self):
         return f"<DonorsOverview ({self.rodne_cislo})>"
 
+    @staticmethod
+    def search(search_str):
+        """Returns a filter to pass to `query.filter()` for a given search string"""
+        conditions_all = [
+            [
+                DonorsOverview.rodne_cislo.like(f"%{part}%"),
+                DonorsOverview.first_name.like(f"%{part}%"),
+                DonorsOverview.last_name.like(f"%{part}%"),
+                DonorsOverview.address.like(f"%{part}%"),
+                DonorsOverview.city.like(f"%{part}%"),
+                DonorsOverview.postal_code.like(f"%{part}%"),
+                DonorsOverview.kod_pojistovny.like(f"%{part}%"),
+            ]
+            for part in search_str.split(" ")
```

Comment:
Co se stane, když je ve vyhledávacím řetězci těch mezer víc? Jakože `"Jan  Novák"?

Reply:
Bude to fungovat jako s jednou mezerou:
```sql
    (jmeno LIKE %Jan% OR prijmeni = %Jan% ...)
AND (jmeno LIKE %% OR prijmeni LIKE %% ...)
AND (jmeno LIKE %Novak% OR prijmenl LIKE %Novak% ...)
```

Přibyde jedna podmínka, která bude vždy pravdivá, ale stále budou muset být splněny zbylé dvě. Můžu na to přidat speciální případ, ale myslím si, že zrychlení, které bychom tím získali, by bylo zcela zanedbatelné.

Reply:
Nebo nerozdělovat ten řetězec podle mezer, nýbrž podle souvislé řady bílého místa `/\s+/`

Reply:
Co se stane, když bude _search_str_ obsahovat procento?

Reply:
> Nebo nerozdělovat ten řetězec podle mezer, nýbrž podle souvislé řady bílého místa search_str.split().
Díky, nevěděl jsem, že to jde.

> Nezapomeň zdvojit procento před vložením do LIKE.
`like()` jsem změnil na funkci `contains()`, která by to měla dělat automaticky.

Reply:
To nahrazování prý ne.

> . LIKE wildcard characters % and _ are not escaped by default unless the ColumnOperators.contains.autoescape flag is set to True.